### PR TITLE
[PUBDEV-4546] In logs, we can se PID to be set to wrong value before …

### PR DESCRIPTION
**Pull request by jakubhava**
…it's properly initialized

We need to initialize PID before we initialize SELF_ADDRESS. Initialization of SELF_ADDRESS triggers buffered logged messages to be logged and PID is part of the log header. We want to have correct PID in the buffered messaged as well

This is the last but one issue in the logs I found.

### CODE REVIEW

**Comment by abal5** [comment]
@h2o-ops please test!

**Comment by jakubhava** [comment]
@h2o-ops please test!

File: h2o-core/src/main/java/water/H2O.java [hunk 1]
```diff
@@ -2139,6 +2137,16 @@ public static void main( String[] args ) {
     Log.debug("    Start GA: " + (time12 - time11) + "ms");
   }
 
+  /** Find PID of the current process, use -1 if we can find the value */
+  private static void initializePID(){
```
Comment (michalkurka) [inline h2o-core/src/main/java/water/H2O.java]: Suggestion: remove the side effect of this method; instead changing the static field PID directly return the pid as a result of the method. Set the PID either in main method or in a static initialization block.

Comment (jakubhava) [inline h2o-core/src/main/java/water/H2O.java]: Good point. I wanted to do the smallest possible change, however why not improve right away. Thanks! 👍

**Pull request merged**